deeplinkdispatch-base/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.kt

```diff
@@ -76,19 +76,34 @@ data class DeepLinkMatchResult(val deeplinkEntry: DeepLinkEntry,
     }
 }
 
-data class DeepLinkEntry(val uriTemplate: String,
-                         /**
-                          * The class where the annotation corresponding to where an instance of DeepLinkEntry is declared.
-                          */
-                         val activityClass: Class<*>,
-                         val method: String?) {
+data class DeepLinkEntry(
+    val uriTemplate: String,
+    /**
+     * The class name where the annotation corresponding to where an instance of DeepLinkEntry is declared.
+     */
+    val className: String,
+    val method: String?
+) {
 
     enum class Type {
         CLASS,
         METHOD
     }
 
+    val activityClass: Class<*>? by lazy {
+        try {
+            Class.forName(className)
+        } catch (e: ClassNotFoundException) {
+            println(
```

Comment:
I'm a bit rusty with how DLD error handling works, but it seems like it would be ideal to have this error message bubbled up to the error handler. otherwise  this root cause will easily go missed right?

Reply:
@elihart this is not a Android module but a Java module, so we do not have access here.

In reality this class is read from a class that is annotated (so it will exist) but it could not be in class loader scope (e.g. if it is in a dynamic feature). I'll have a second look at the error handler but IMHO that is sadly the best we can do from here without bigger changes.

Reply:
@elihart I did update this and this is now in addition also reported via the error handler.